src/circuit.rs

```diff
@@ -92,6 +92,91 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An assigned cell.
+#[derive(Clone, Copy, Debug)]
+pub struct AssignedCell<F: Field, T: Copy + Into<F>> {
+    value: Option<T>,
+    cell: Cell,
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field, T: Copy + Into<F>> AssignedCell<F, T> {
+    /// Assign a new AssignedCell.
+    pub fn assign<A, AR>(
```

Comment:
Perhaps this should be called `assign_unchecked` (or whatever) so that if you have an `AssignedCell<F, T>` you aren't misled into thinking that the circuit has constrained the cell to be a value of `T` (which this code does not do). The result is that reviewers of circuit code will be more careful when looking at `assign_unchecked` code, so it'll be better to isolate its usage in abstractions like `AssignedBool` and so on.